### ISSUE TITLE
Labwc icon to ship within the main repo for its labwc.desktop session file

### ISSUE THREAD

**Issue by Consolatis** [labwc/labwc-artwork#7]
Ref:
- https://github.com/labwc/labwc/pull/1462

Lets brainstorm here regarding the proper dimensions / scaling / transparency.
What we need is a squared canvas with a properly sized labwc logo, I think the default is 256x256 but not sure about that.

**Comment by redtide** [comment]
I believe that if you use a SVG it doesn't matter the image size, maybe just the proportion for the margins when scaled? Not sure.
I did [some work](https://github.com/labwc/labwc-artwork/assets/1171962/634baf74-437a-487f-a217-2c9307c7af3d) last night with Inkscape and SVGOMG today to reduce it a bit (left some detail in the image comments). If you are going to use it, feel free to modify it as required.

**Comment by domo141** [comment]
Good stuff redtide -- I could not resist the temptation and edit it a bit w/ a text editor:

```
<!--Modified by redtide with Inkscape (http://www.inkscape.org/) and SVGOMG (https://jakearchibald.github.io/svgomg/)-->
<!--labwc logo squared (C) Johan Malm LICENSE: CC BY-SA 4.0-->
<svg xmlns="http://www.w3.org/2000/svg" width="256mm" height="256mm"
     viewBox="0 0 114 114"
     stroke-linecap="round" stroke-linejoin="round" stroke-width="6">
  <path fill="#f0d70f" stroke="#f0d70f" d="m 6 18 46 27 v 55 L 13 69 Z"/>
  <path fill="#d02f90" stroke="#d02f90" d="m 108 18 -47 27 v 55 l 39 -30 z"/>
</svg>
```

**Comment by redtide** [comment]
> Good stuff redtide -- I could not resist the temptation and edit it a bit w/ a text editor:
> 
> ```
> 
> 
> <svg xmlns="http://www.w3.org/2000/svg" width="256mm" height="256mm"
>      viewBox="0 0 114 114"
>      stroke-linecap="round" stroke-linejoin="round" stroke-width="6">
>   <path fill="#f0d70f" stroke="#f0d70f" d="m 6 18 46 27 v 55 L 13 69 Z"/>
>   <path fill="#d02f90" stroke="#d02f90" d="m 108 18 -47 27 v 55 l 39 -30 z"/>
> </svg>
> ```

Thank you! But now it misses a "Simplified and optimized by domo141" :)

**Comment by Consolatis** [comment]
For comparison:
Orga logo (png 5.2 KiB):
![labwc_orga_logo](https://github.com/labwc/labwc-artwork/assets/35009135/da0657aa-287c-477e-a164-e0800b774d7b)

labwc2.svg from this repo (svg 1.2 KiB):
![labwc2](https://github.com/labwc/labwc-artwork/assets/35009135/ac842fad-4d48-4ded-b667-411ccbc83055)

l2.svg created by @domo141 (svg 584 Bytes)
![l2](https://github.com/labwc/labwc-artwork/assets/35009135/af683c3d-c8a4-421d-ab99-1a636a94cf4e)

labwc_redtide_1 from post above (svg, 558 Bytes)
![labwc_redtide_1](https://github.com/labwc/labwc-artwork/assets/35009135/e108ef79-f94c-44d9-9567-1d6f2c512a0a)

I think the main issue is that `labwc_redtide_1` centers the logo compared to the orga logo that offsets it a bit to the bottom which makes sense as the implied invisible top pane of the cube is also part of the logo. Otherwise I think that one would be pretty perfect.

**Comment by 01micko** [comment]
This is a _complete_ nit-pick (OCD methinks) but I dislike mixing absolute and relative svg paths.
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="256px" height="256px"
     viewBox="0 0 114 114"
     stroke-linecap="round" stroke-linejoin="round" stroke-width="6">
  <path fill="#f0d70f" stroke="#f0d70f" d="m 6 18 46 27 v 55 l -39 -30 z"/>
  <path fill="#d02f90" stroke="#d02f90" d="m 108 18 -47 27 v 55 l 39 -30 z"/>
</svg>
```

**Comment by domo141** [comment]
01micko: I like that -- tried but failed with that L -> l change (due not understood to change that -39 -30 at the end :D

**Comment by 01micko** [comment]
> centers the logo

With relpaths it's a simple matter to drop the whole image 10 units (in my case `px`)

**Comment by domo141** [comment]
yes

simple expnanation of the first path above: (m 6,18 46,27 v 55 l -39,-30)
- move to coordinates 6.18
- implicit lineto to 46,27 (46 right, 27 down)
- vertical lineto 55 down
- lineto 39 left, 30 up
- z closepath's

(and the second, purple path, circles counterclockwise)

(for symmetry probably that 46 and -47 should have same abs value)

**Comment by redtide** [comment]
For some reason I thought it was a colored book, not a cube 😶‍🌫️

**Comment by Consolatis** [comment]
Maybe its both :)

**Comment by domo141** [comment]
More funplay: viewbox="0 0 256 256" (and coordinates scaled to that), for easier "understanding"
```
<?xml version="1.0" encoding="UTF-8"?>
<!--labwc logo #2 (C) Johan Malm LICENSE: CC BY-SA 4.0-->
<svg xmlns="http://www.w3.org/2000/svg" width="256px" height="256px"
     viewbox="0 0 256 256"
     stroke-linecap="round" stroke-linejoin="round" stroke-width="14">
  <path fill="#f0d70f" stroke="#f0d70f" d="m 11 41 105 61 v 123 l -87 -68 z" />
  <path fill="#d02f90" stroke="#d02f90" d="m 244 41 -105 61 v 123 l 87 -68 z" />
</svg>
```

**Comment by Consolatis** [comment]
Logo by @domo141 posted at https://github.com/labwc/labwc.github.io/pull/42#issuecomment-1925425886:
![labwc_logo_domo](https://github.com/labwc/labwc-artwork/assets/35009135/79ca62f4-207a-42ef-86ed-a76e5f461f9e)

SVG:
```svg
<?xml version="1.0" encoding="UTF-8"?>
<!--labwc logo #2 (C) Johan Malm LICENSE: CC BY-SA 4.0-->
<svg xmlns="http://www.w3.org/2000/svg" width="256px" height="256px"
     viewbox="0 0 256 256"
     stroke-linecap="round" stroke-linejoin="round" stroke-width="12">
  <path fill="#f0d70f" stroke="#f0d70f" d="m 26 68 91 55 v 108 l -76 -61 z" />
  <path fill="#d02f90" stroke="#d02f90" d="m 229 68 -91 55 v 108 l 76 -61 z" />
</svg>
```

Personally I think it looks pretty great. Thanks to everybody contributing here!
Up to @johanmalm as the project lead to decide on the actual logo to use as icon.

**Comment by domo141** [comment]
...and as an original author

**Comment by ludgie** [comment]
@domo141 you're the original author of the cube logo?

**Comment by domo141** [comment]
No, not me, @johanmalm - sorry for the confusion...

**Comment by johanmalm** [comment]
> 
> SVG:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> 
> <svg xmlns="http://www.w3.org/2000/svg" width="256px" height="256px"
>      viewbox="0 0 256 256"
>      stroke-linecap="round" stroke-linejoin="round" stroke-width="12">
>   <path fill="#f0d70f" stroke="#f0d70f" d="m 26 68 91 55 v 108 l -76 -61 z" />
>   <path fill="#d02f90" stroke="#d02f90" d="m 229 68 -91 55 v 108 l 76 -61 z" />
> </svg>
> ```
> 
> Personally I think it looks pretty great. Thanks to everybody contributing here! Up to @johanmalm as the project lead to decide on the actual logo to use as icon.

I like this one by @domo141 and it seems incredible close to the original one (https://github.com/labwc/labwc-artwork/blob/main/logos/labwc2.svg) but has much simpler XML code which is nice.

I suggest putting it at `data/labwc.svg`

Does anyone feel like doing the honours of a PR? If not, I can sort next week.

As next steps I think we should:

1. Create a new `labwc2-with-text.svg` based on the above. Let's settle the font discussion first though.
2. Move `docs/labwc.desktop` to `data/` too.
3. Add a `data/labwc-symbolic.svg`
4. Adjust `meson.build`
5. Remove the equivalent files from `logos/`

**Comment by rcalixte** [comment]
> I suggest putting it at `data/labwc.svg`
> 
> Does anyone feel like doing the honours of a PR? If not, I can sort next week.
> 
> As next steps I think we should:
> 
>     2. Move `docs/labwc.desktop` to `data/` too.
> 
>     3. Add a `data/labwc-symbolic.svg`
> 
>     4. Adjust `meson.build`

I've adjusted the PR at labwc/labwc#1462 so it should be ready to review now. (I just saw it was merged.)


>     5. Remove the equivalent files from `logos/`

I can create a PR for this and link it to the aforementioned PR too. I had a question of if the `labwc.svg` in the root should be removed as well @johanmalm.

**Comment by johanmalm** [comment]
> > ```
> > 5. Remove the equivalent files from `logos/`
> > ```
> 
> I can create a PR for this and link it to the aforementioned PR too. I had a question of if the `labwc.svg` in the root should be removed as well @johanmalm.

Great.

Regarding the `labwc.svg` in the root directory, yes, I think that should be removed now, but we ought to check with @01micko because they created that one.

**Comment by Consolatis** [comment]
We could also rename the variants from the main repo to `labwc_icon.svg` / `labwc_icon-symbolic.svg` in this repo. We don't really need to have a single combined file for all use-cases (e.g. https://labwc.github.io, screenshots, icon and so on).

**Comment by 01micko** [comment]
No qualms here. :)

**Comment by 01micko** [comment]
I think this is closed by #9 . Feel free to re-open if necessary.

Also in 758ae11 I updated the logo in the `labwcmoon.ttf` font. A little demo follows.. (made with [mkwallpaper](https://github.com/01micko/mkwallpaper) )

![labwc-wall6](https://github.com/labwc/labwc-artwork/assets/1019119/cbf1273a-21e3-4fe0-9b14-4673ea21de98)
